lib/dao.js

```diff
@@ -2407,6 +2577,201 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replace
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+DataAccessObject.prototype.replaceAttributes = function(data, options, cb) {
+  var Model = this.constructor;
+  var id = getIdValue(this.constructor, this);
+  var idValue = idName(this.constructor)
+  data[idValue] = id;
```

Comment:
It may not look neat; I added id field and its value to data, because in line 2602 data should have an `id` when creating instance:
```
var inst = new Model(data);
```
And then deleted the id field from data on line 2610:
```
delete data[idValue];
```

So it is probably ugly!?